tests/pubsub/check_pubsub_connection_xdp.c

```diff
@@ -0,0 +1,285 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Copyright (c) 2019-2020 Kalycito Infotech Private Limited
+ */
+
+#include <open62541/server.h>
+#include <open62541/server_config_default.h>
+#include <ua_server_internal.h>
+#include <open62541/types_generated_encoding_binary.h>
+#include <open62541/types_generated.h>
+#include <open62541/plugin/pubsub_ethernet.h>
+#include <open62541/plugin/pubsub_ethernet_xdp.h>
+#include "ua_pubsub.h"
+
+#include <linux/if_packet.h>
+#include <netinet/ether.h>
+
+#include <linux/bpf.h>
+#include <linux/if_link.h>
+#include "/usr/local/src/bpf-next/include/uapi/linux/if_xdp.h"
+#include "/usr/local/src/bpf-next/tools/testing/selftests/bpf/bpf_util.h"
+#include "/usr/local/src/bpf-next/tools/lib/bpf/bpf.h"
+#include "/usr/local/src/bpf-next/tools/lib/bpf/libbpf.h"
+#include "/usr/local/src/bpf-next/samples/bpf/xdpsock.h"
+#include <sys/resource.h>
+#include <sys/mman.h>
+
+#include <check.h>
+
+#define             ETHERNET_INTERFACE                 "enp2s0"
```

Comment:
Such system-specific hard-coded values are problematic for a unit test.
This will be a nightmare to maintain in an automated build pipeline.
Can this configuration be detected automatically?
Otherwise we can read it from an environment variable.

Reply:
Modified ETHERNET_INTERFACE value to be taken from command line argument.